app/src/main/java/org/connectbot/transport/SSH.java

```diff
@@ -439,6 +440,7 @@ private void finishConnection() {
 
 	@Override
 	public void connect() {
+		connecting = true;
```

Comment:
This line didn't change with the addition of the intermediate function, but it is testing out ok on my build.